Mathlib/Data/Finsupp/Defs.lean

```diff
@@ -326,25 +330,26 @@ theorem single_eq_pi_single [DecidableEq α] (a : α) (b : M) : ⇑(single a b)
 
 @[simp]
 theorem single_zero (a : α) : (single a 0 : α →₀ M) = 0 :=
-  coe_fn_injective <| by
-    simpa only [single_eq_update, coe_zero] using Function.update_eq_self a (0 : α → M)
+  coeFn_injective <| by
+    classical simpa only [single_eq_update, coe_zero] using Function.update_eq_self a (0 : α → M)
 #align finsupp.single_zero Finsupp.single_zero
 
 theorem single_of_single_apply (a a' : α) (b : M) :
     single a ((single a' b) a) = single a' (single a' b) a := by
+  classical
   rw [single_apply, single_apply]
   ext
-  split_ifs
+  split_ifs with h
   · rw [h]
-  · rw [zero_apply, single_apply, if_t_t]
+  · rw [zero_apply, single_apply, ite_self]
 #align finsupp.single_of_single_apply Finsupp.single_of_single_apply
 
 theorem support_single_ne_zero (a : α) (hb : b ≠ 0) : (single a b).support = {a} :=
   if_neg hb
 #align finsupp.support_single_ne_zero Finsupp.support_single_ne_zero
 
-theorem support_single_subset : (single a b).support ⊆ {a} :=
-  show ite _ _ _ ⊆ _ by split_ifs <;> [exact empty_subset _, exact subset.refl _]
+theorem support_single_subset : (single a b).support ⊆ {a} := by
+  classical show ite _ _ _ ⊆ _; split_ifs <;> [exact empty_subset _, exact Subset.refl _]
```

Comment:
We might want to wait for https://github.com/leanprover-community/mathlib/pull/18275/files so that this type of thing is fixed in Mathlib3